src/SampleChainsDynamicHMC.jl

```diff
@@ -54,17 +54,55 @@ function SampleChains.step!(chain::DynamicHMCChain)
     Q, tree_stats = DynamicHMC.mcmc_next_step(getfield(chain, :meta), getfield(chain, :state))
 end
 
-function SampleChains.initialize!(rng::Random.AbstractRNG, ::Type{DynamicHMCChain}, ℓ, tr, ad_backend=Val(:ForwardDiff))
+@concrete struct DynamicHMCConfig <: ChainConfig{DynamicHMCChain}
+    init
+    warmup_stages
+    algorithm
+    reporter
+end
+
+# Docs adapted from https://tamaspapp.eu/DynamicHMC.jl/stable/interface/
+"""
+    dynamicHMC(
+        init          = ()
+      , warmup_stages = DynamicHMC.default_warmup_stages()
+      , algorithm     = DynamicHMC.NUTS()
+      , reporter      = DynamicHMC.NoProgressReport()
+    )
+
+`init`: a NamedTuple which can contain the following fields (all of them optional and provided with reasonable defaults):
+- q: initial position. Default: random (uniform [-2,2] for each coordinate).
+- κ: kinetic energy specification. Default: Gaussian with identity matrix.
+- ϵ: a scalar for initial stepsize, or nothing for heuristic finders.
+
+`warmup_stages`: a sequence of warmup stages. See `default_warmup_stages` and `fixed_stepsize_warmup_stages`; the latter requires an ϵ in initialization.
+
+`algorithm`: see `NUTS`. It is very unlikely you need to modify this, except perhaps for the maximum depth.
+
+`reporter`: how progress is reported. By default, verbosely for interactive sessions using the log message mechanism (see `LogProgressReport`, and no reporting for non-interactive sessions (see `NoProgressReport`).
+
+For more details see https://tamaspapp.eu/DynamicHMC.jl/stable/interface/
+"""   
```

Comment:
This example would be helpful for users to know how to change common arguments:

```suggestion
# Example

```jldoctest
julia> using LinearAlgebra

julia> config = dynamicHMC(
           warmup_stages=default_warmup_stages(
               M=Symmetric, # adapt dense positive definite metric
               stepsize_adaptation=DualAveraging(δ=0.9), # target acceptance rate 0.9
               doubling_stages=7, # 7-stage metric adaptation
           ),
           reporter=LogProgressReport(), # log progress using Logging
       );
```
"""
```